src/binaryutils.cpp

```diff
@@ -0,0 +1,259 @@
+    const cv::Mat input) {
+    // Network and input must have the same number of shape dimensions. 
+    if ((unsigned int)(input.channels()) != tensorInfo->getShape()[1]) {
+        SPDLOG_DEBUG("Binary sent to input: {} has invalid number of channels. Expected: {} Actual: {}", tensorInfo->getMappedName(),tensorInfo->getShape()[1], input.dims);
+        return Status(StatusCode::INVALID_NO_OF_SHAPE_DIMENSIONS);
+    }
+    return StatusCode::OK;
+}
+
+const bool checkBatchSizeMismatch(const std::shared_ptr<TensorInfo>& tensorInfo,
+    const int batchSize) {
+    if (static_cast<size_t>(batchSize) != tensorInfo->getShape()[0])
+        return true;
+    return false;
+}
+
+const Status validateInput(const std::shared_ptr<TensorInfo>& tensorInfo,
+    const cv::Mat input) {
+    auto status = validateNumberOfChannels(tensorInfo, input);
+    if (!status.ok())
+        return status;
+
+    return StatusCode::OK;
+}
+
+const Status validateTensor(const std::shared_ptr<TensorInfo>& tensorInfo,
+    const tensorflow::TensorProto& src) {
+    if(tensorInfo->getLayout() == InferenceEngine::Layout::NCHW ||
+       tensorInfo->getLayout() == InferenceEngine::Layout::NHWC)
+    {
+        return StatusCode::OK;
+    }
+
+    if (checkBatchSizeMismatch(tensorInfo, src.string_val_size())){
+        SPDLOG_DEBUG("Input: {} request batch size is incorrect. Expected: {} Actual: {}", tensorInfo->getMappedName(), tensorInfo->getShape()[0],src.string_val_size());
+        return Status(StatusCode::UNSUPPORTED_LAYOUT);
+    }
+
+    return StatusCode::OK;
+}
+
+Status convertTensorToMatsMatchingTensorInfo(const tensorflow::TensorProto& src, std::vector<cv::Mat>& images, const std::shared_ptr<TensorInfo>& tensorInfo) {
+
+    for (int i = 0; i < src.string_val_size(); i++) {
+        cv::Mat image = convertStringValToMat(src.string_val(i));
+        if (image.data == NULL)
+            return StatusCode::IMAGE_PARSING_FAILED;
+                    
+        auto status = validateInput(tensorInfo, image);
+        if (status != StatusCode::OK) {
+            return status;
+        }
+
+        if (!isPrecisionEqual(image.depth(), tensorInfo->getPrecision())) {
+            cv::Mat imagePrecisionConverted;
+            status = convertPrecision(image, imagePrecisionConverted, tensorInfo->getPrecision());
+
+            if (status != StatusCode::OK) {
+                return status;
+            }
+
+            cv::Mat imageResized;
+            resizeMat(imagePrecisionConverted, imageResized, tensorInfo);
+            if (status != StatusCode::OK) {
+                return status;
+            }
+
+            images.push_back(imageResized);
+        } else {
+            cv::Mat imageResized;
+            resizeMat(image, imageResized, tensorInfo);
+            if (status != StatusCode::OK) {
+                return status;
+            }
+            images.push_back(imageResized);
+        }
```

Comment:
I think this block can be a bit shorter. The `resizeMat` call is performed almost identically in both paths.
Could you try to go with something like:
```
cv::Mat imagePostValidation = image;
if (!isPrecisionEqual(image.depth(), tensorInfo->getPrecision())) {
      status = convertPrecision(image, imagePostValidation, tensorInfo->getPrecision());

      if (status != StatusCode::OK) 
           return status;
}
cv::Mat imageResized;
resizeMat(imagePostValidation, imageResized, tensorInfo);
if (status != StatusCode::OK)
         return status;
images.push_back(imageResized);
```

Or even not use imagePostValidation variable if its possible to perform operations in-place like:
`status = convertPrecision(image, image, tensorInfo->getPrecision());`